src/plugins/visualizations/public/legacy/build_pipeline.ts

```diff
@@ -258,49 +227,49 @@ const adjustVislibDimensionFormmaters = (vis: Vis, dimensions: { y: any[] }): vo
 };
 
 export const buildPipelineVisFunction: BuildPipelineVisFunction = {
-  vega: params => {
-    return `vega ${prepareString('spec', params.spec)}`;
+  vega: ({ spec }) => {
```

Comment:
all of these should move to specific charts toExpression function, can be done in followup